src/algorithm/pso.cpp

```diff
@@ -565,6 +567,51 @@ void pso::initialize_topology__adaptive_random( std::vector< std::vector<int> >
 }
 
 
+/**
+ * @brief Arranges particles in a 2D lattice with alternatively staggered rows, where each interacts with its 6 nearest neighbours.
+
+*/
+const int hexagonal_neighb_diff_even[6][2] = { {-1,-1}, {0,-1}, {-1,0}, {1,0}, {-1,1}, {0,1} };
+const int hexagonal_neighb_diff_odd[6][2] = { {0,-1}, {1,-1}, {-1,0}, {1,0}, {0,1}, {1,1} };
+
+void pso::initialize_topology__hexagonal( std::vector< std::vector<int> > &neighb ) const
+{
+        int swarm_size = neighb.size();
+        int     cols, rows;             // lattice structure
+        int     pidx, nidx;             // particle and neighbour indices, in the swarm and neighbourhood vectors
+        int     p_x, p_y;               // particle's coordinates in the lattice
+        int     n_x, n_y;               // particle neighbor's coordinates in the lattice
+
+        rows = std::sqrt( swarm_size );
+        while( ( (swarm_size % rows != 0) || (rows % 2 == 1)) && (rows>1)  )
```

Comment:
Spaces around operator ``>``. You have an extra space before last parenthesis. (I suggest installing some C++/Python coding style checker.